spec/index_spec.rb

```diff
@@ -76,7 +76,7 @@
   context "#[]" do
     before do
       @id = Daru::Index.new [:one, :two, :three, :four, :five, :six, :seven]
-      @mixed_id = Daru::Index.new ['a','b','c',:d,:a,0,3,5]
+      @mixed_id = Daru::Index.new ['a','b','c',:d,:a,8,3,5]
```

Comment:
Example 1. Notice that since `0` is present in the index, daru should not return the _0th element_ but rather the element at index `0`. This applies for ranges too.
``` ruby
[18] pry(main)> a = Daru::Vector.new([1,2,3,4], index: [2,3,0,1])
=> 
#<Daru::Vector:88293040 @name = nil @size = 4 >
    nil
  2   1
  3   2
  0   3
  1   4

[19] pry(main)> a[0]
=> 3

```

Example 2. Here 0 isn't present in the index so it returns the _0th element_:
``` ruby

[24] pry(main)> b = Daru::Vector.new([4,3,5,9], index: [:a, :b, :c, :d])
=> 
#<Daru::Vector:87873110 @name = nil @size = 4 >
    nil
  a   4
  b   3
  c   5
  d   9

[25] pry(main)> b[0]
=> 4

```